src/IssueViz.Security/UI/HotspotsList/HotspotsControl.xaml

```diff
@@ -4,223 +4,61 @@
-                                BorderThickness="0" 
-                                Margin="2"
-                                Padding="3" 
-                                SnapsToDevicePixels="true">
-                            <Grid>
-                                <Grid.ColumnDefinitions>
-                                    <ColumnDefinition MinWidth="25" Width="Auto" />
-                                    <ColumnDefinition Width="*"/>
-                                </Grid.ColumnDefinitions>
-                                <ContentPresenter Grid.Column="0" x:Name="Icon" Margin="5,0,10,0" VerticalAlignment="Center" ContentSource="Icon"/>
-                                <ContentPresenter Grid.Column="1" x:Name="Header" ContentSource="Header" />
-                            </Grid>
-                        </Border>
-                        <ControlTemplate.Triggers>
-                            <Trigger Property="IsHighlighted" Value="True">
-                                <Setter Property="Background"  Value="{DynamicResource {x:Static vsShell:VsBrushes.CommandBarHoverOverSelectedKey}}"/>
-                            </Trigger>
-                        </ControlTemplate.Triggers>
-                    </ControlTemplate>
-                </Setter.Value>
-            </Setter>
-        </Style>
-        <Style x:Key="PriorityColumnCellTextBlockStyle" TargetType="TextBlock">
-            <Setter Property="TextWrapping" Value="Wrap"/>
-            <Setter Property="TextAlignment" Value="Center"/>
-            <Setter Property="VerticalAlignment" Value="Center"/>
-            <Setter Property="Margin" Value="1"/>
-            <Setter Property="Background" Value="{Binding Path=DataContext.Hotspot.Issue.Rule.Priority, RelativeSource={RelativeSource Self}, Converter={StaticResource PriorityToBackgroundConverter}}"/>
-        </Style>
+            <ContextMenu x:Key="RowMenu" DataContext="{Binding PlacementTarget.DataContext, RelativeSource={RelativeSource Self}}">
+                <MenuItem Header="Remove" 
+                    CommandParameter="{Binding}"
+                    Command="{Binding DataContext.RemoveCommand, RelativeSource={RelativeSource Mode=FindAncestor, AncestorType=DataGrid}}">
+                    <MenuItem.Icon>
+                        <vsImaging:CrispImage Moniker="{x:Static vsCatalog:KnownMonikers.DeleteTableRow}"/>
+                    </MenuItem.Icon>
+                </MenuItem>
+            </ContextMenu>
+            
+            <!-- Extend the shared data grid row style to add the context menu and navigability data bindings -->
+            <Style BasedOn="{StaticResource BaseDataGridRow}"
+                   TargetType="DataGridRow">
+                <Setter Property="ContextMenu" Value="{StaticResource RowMenu}" />
+                <Setter Property="Background" Value="{DynamicResource {x:Static vsTheming:TreeViewColors.BackgroundBrushKey}}"/>
```

Comment:
The background has the same definition as in the base style, which shouldn't be necessary. However, if I miss it out the bits of the row that aren't covered by cells aren't coloured correctly. I don't know why.

I left the background in the base style so the base style will work correctly if it isn't overridden.

Reply:
Hm, I had something similar. Can you please retry by reloading VS? I think it might be a xaml cache.

Reply:
"Reloading VS"? Do you mean starting a new debug session? If so, I tried that.

Reply:
Yes that what I meant. Oh well :(